src/tutorial/microbe_stage/MicrobeMovement.cs

```diff
@@ -84,28 +88,19 @@ public override void ApplyGUIState(MicrobeTutorialGUI gui)
             return false;
         }
 
-        protected override void OnProcess(TutorialState overallState, float delta)
+        [RunOnAxis(new[] { "g_move_forward", "g_move_backwards" }, new[] { -1.0f, 1.0f })]
+        [RunOnAxis(new[] { "g_move_left", "g_move_right" }, new[] { -1.0f, 1.0f })]
+        [RunOnAxisGroup]
```

Comment:
Is this attribute needed? I think it would be simpler to use the system if specifying RunOnAxis attributes would automatically make the input manager handle this as if RunOnAxisGroup was specified.

Reply:
I see `[RunOnAxisGroup(InvokeWithNoInput = true)]` is used elsewhere, but when not needing any special settings, I think that it would be easier for future programmers if the input system assumed that RunOnAxis means default RunOnAxisGroup if that attribute is not specifically added.

Reply:
Without the `RunOnAxisGroup` this would mean, that this Method has two `RunOnAxisAttribute`s attached to it. When either one of those axes is triggered, the method will be called with `(float delta, float value)`. With `RunOnAxisGroup` the system knows that theese represent different axes and therefore call `(float delta, float value1, float value2)`.
In a nutshell: you can attach multiple `InputAttributes` to a method. Therefore it is valid to attach multiple `RunOnKeyDownAttribute`s or multiple `RunOnAxisAttribute`s. The problem is, that you then can't differenciate which key or axes triggered this method.

It is very unlikely that this will ever be utilized, but the system supports it. I hope you get what I mean

Reply:
One example for multiple InputAttributes would be, if you had
- slow zoom in
- slow zoom out
- fast zoom in
- fast zoom out

These two implementations would be valid:
```
[RunOnAxis(new[] { "g_zoom_out_slow", "g_zoom_in_slow", "g_zoom_out_fast", "g_zoom_in_fast" }, new[] { -1, 1, -3, 3 })]
```

```
[RunOnAxis(new[] { "g_zoom_out_slow", "g_zoom_in_slow" }, new[] { -1, 1 })]
[RunOnAxis(new[] { "g_zoom_out_fast", "g_zoom_in_fast" }, new[] { -3, 3 })]
```

Reply:
I now seem to remember something about that. Could you write more xml comments, and maybe write a short overview like "doc/input_system.md" file. This is a pretty important system for new programmers *and* me to understand.